docs/report_stage_i.md

```diff
@@ -0,0 +1,497 @@
+
+Instead, we will present the questions that will be especially interesting to answer in the continuation of this research.
+
+#### Q1: Can we devise a one-size-fits-all set of OPCODE gas costs?
+
+After visually comparing the implementation-relative measurements (see **TODO reference** above), we suspect that at the current state of EVM/eWASM implementations, it might be challenging to propose gas costs to OPCODEs, which would apply equally well to all.
+
+E.g., from the preliminary measurement data we saw that, for `evmone` and `openethereum`, several arithmetic OPCODEs (range from `DIV` to `MULMOD`) are consequently more expensive to execute than the pivot OPCODE.
+This is however not the case for `geth`.
+This means that, given these results are accurate, which is to be further confirmed in Stage II, we might not expect a single good set of gas costs for all the OPCODEs in question, unless there is a parallel effort made to optimize the costly OPCODEs in selected implementations.
+
+#### Q2: is measuring of individual instructions feasible?
+
+The obtained results led us to the conclusion that measuring of individual instructions can be feasible.
+However, care must be taken to perform validations of the assumption that the lack of timer precision or its overhead, nor the inclusion of timer function calls itself, don't introduce "unfairness" to the measurements collected.
+Refer to the validation techniques planned for Stage II analysis in **TODO ref section approach etc** for ideas on tackling this.
+There, we will also list possible fallback plans in case there is evidence that the noise and overhead present in this method of measurement is unacceptable.
+
+#### Q3: How can we explore the entire program space to capture all sources of variability of OPCODEs computational cost
+
+The main intention of the research presented here is to obtain relative comparison of costs of OPCODEs, accounting for the sources of variability that are already represented by the current gas cost schedule.
+An example of such variability is that the gas cost of a `SHA3` OPCODE is a function of the size of its arguments, which naturally impact the computational cost of the operation.
+
+It is tempting to search for sources of variability unaccounted for in the current gas schedule.
+A hypothetical example of such situation would be one, where, for example, `PUSH1` operation became more expensive, after having been repeated multiple times in the program.
+
+#### Q4: How should warm-up be treated?
+
+Preliminary results indicate that first OPCODEs to execute in the program, as well as first OPCODEs to execute in the context of an OS process, have a noticeable penalty in terms of execution duration.
+This is a natural phenomenon usually referred to as warm-up in program benchmarking, and the usual treatment is discarding of the first batch of measurement samples.
```

Comment:
Perhaps  we should also discuss the influence of caching/locality. In real life EVM seldom executes exactly te same code in short sequence.